managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -3406,6 +3411,9 @@ public LongPairRangeSet<PositionImpl> getIndividuallyDeletedMessagesSet() {
     }
 
     public boolean isMessageDeleted(Position position) {
+        if (this instanceof ReadOnlyCursor) {
+            return false;
+        }
```

Comment:
It's better to overwrite the method in ReadOnlyCursorImpl.java

    @Override
    public boolean isMessageDeleted(Position position) {
        return false; 
    }

Reply:
Yes, override is better